examples/lib/pods_test.rego

```diff
@@ -0,0 +1,91 @@
+package lib.pods
+
+test_input_as_other {
+    input := {
+        "kind": "Other",
+        "spec": {
+            "containers": [{}]
+        }
+    }
+
+    resource := pod with input as input
+
+    not resource
+}
+
+test_input_as_pod {
+    input := {
+        "kind": "Pod",
+        "spec": {
+            "containers": [{}]
+        }
+    }
+
+    resource := pod with input as input
+
+    resource.spec.containers
+}
+
+test_input_as_deployment {
```

Comment:
nit: Perhaps we should rename this test? We are testing more than deployments since most of our `Kind`s match this.

Reply:
Yeah I wasn't sure what else to call it. I went for the consistency play because everything else follows "as __" where the ___ is the kind.